Articles/HowTos/CriticalBeginnerGitUsageTips.md

```diff
@@ -0,0 +1,99 @@
+#### Publication date: May ??, 2021
+
+**A) Set up minimal global settings right away on every new computer.**
+Always set up a consistent global Git `user.name` and `user.email` on every machine where you use Git right away.
+(Otherwise you will show up as many different developers according to Git, and it is a pain to resolve these in all of the Git repos you commit to after the fact).
+Also, disable push of all branches by default (prior to Git 2.0) or a simple `git push` will push all branches.
+In addition, it is nice to see Git output with color (default is no color).
+And, to avoid having to resolve the same merge conflicts multiple times, turn on the `git rerere` feature.
+To set these on a new machine, run:
+
+```
+$ git config --global user.name "First M. Last"          # CRITICAL
+$ git config --global user.email "youremail@someurl.com" # CRITICAL
+$ git config --global color.ui true        # Use color in git output to terminal.
+$ git config --global push.default simple  # Or 'tracking' with older versions of git.
+$ git config --global rerere.enabled 1     # Auto-resolve same conflicts on rebase.
+```
+
+Also consider setting up your bash shell prompt and `git` command-completion by downloading the bash scripts [git-prompt.sh](https://raw.githubusercontent.com/git/git/master/contrib/completion/git-prompt.sh) and [git-completion.bash](https://raw.githubusercontent.com/git/git/master/contrib/completion/git-completion.bash) and add activating them by adding them to your `~/.bash_profile` file with the lines:
+
+``` 
+source ~/git-prompt.sh
+source ~/git-completion.bash
+PS1='[\u@\h \W$(__git_ps1 " (%s)")]\$ '
+```
+
+These bash scripts make using Git on the command-line much easier and more productive.
+
+**B) Create proper commits and commit messages.**
+Creating good commits with good commit messages is important for many reasons.
+Commits should contain a single logical change (see "SEPARATE CHANGES" in [gitworkows(7)](https://www.kernel.org/pub/software/scm/git/docs/gitworkflows.html) and "One Commit per Logical Change Solution" in the [Udacity Git course](https://www.udacity.com/course/version-control-with-git--ud123)).
+Commit messages should have a short summary line (under 50 chars is the target) and an optional longer body separated by a blank line (see guidelines in [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)).
+For example:
+
+```
+Summary line for changes (50 chars is target but a little longer okay)
+
+More detailed explanatory text in the body (wrapped to about 72 chars).
+The blank line separating the summary from the body is critical to make
+commands like 'git log --oneline', 'git rebase -i', and others readable
+and useful.
+
+If you use an issue tracker, put references to them like:
+
+Resolves: #123
+Also see: #456, #789
+```
+
+**C) Create small local "checkpoint" commits then cleanup with 'git rebase -i'
+before pushing.**
```

Comment:
Multiline heading here? Shouldn't this be all one line?

Reply:
Yes